src/lib.rs

```diff
@@ -129,6 +129,57 @@ impl<'a> From<PrivatePkcs8KeyDer<'a>> for PrivateKeyDer<'a> {
     }
 }
 
+impl<'a> TryFrom<&'a [u8]> for PrivateKeyDer<'a> {
+    type Error = &'static str;
+
+    fn try_from(key: &'a [u8]) -> Result<Self, Self::Error> {
+        if key.len() <= 5 || key[0] != 0x30 {
+            return Err("invalid key format");
+        }
+
+        // We skip the tag and the length of the sequence
+        let mut skip_len = 2;
+
+        // We skip additonal length bytes
+        if key[1] >= 0x81 {
```

Comment:
I think this is off-by-one no? The maximum value for a short-form DER encoded length is 128 (0x80).

Reply:
Possible I've confused myself here :-)

>  The maximum value for a short-form DER encoded length is 128 (0x80).

I think I should have said 127 here. I believe the short form encoded length values that are possible are `[0 ... 128)` in range notation.

> I don't think it is off-by-one. If the value is 128 or less we would only need to skip two bytes as assigned in line 141.

I think a value of 128 exactly would be `b1000_0000`, and the leading 1 bit indicates it's not a short form length.

Reply:
> I think a value of 128 exactly would be b1000_0000, and the leading 1 bit indicates it's not a short form length.

but... in practice your code would treat it as a long form length with 0 additional bytes, skipping 2 total. So I think it's a moot point in this context anyway.

It doesn't matter how many times I look at this stuff I always find it tricky...

Reply:
I updated the replacement PR's desc to point to this thread and clarify it wasn't really an off-by-one. :+1: